validator/server_arb/machine_cache.go

```diff
@@ -125,24 +137,25 @@ func (c *MachineCache) populateInitialCache(ctx context.Context, target_step uin
 		if nextMachine.GetStepCount() >= target_step {
 			break
 		}
-		if len(c.machines) >= c.config.TargetMachineCount {
+		err := nextMachine.Step(ctx, c.machineStepInterval)
+		if err != nil {
+			return err
+		}
+		nextMachine.Freeze()
+		if len(c.machines) >= c.config.CachedChallengeMachines {
 			// Double the step interval between machines, which halves the number of machines.
 			var pruned []MachineInterface
 			for i, mach := range c.machines {
 				// If i%2 == 0, this machine is no longer on the step interval.
-				if i%2 == 1 {
+				if i%2 == 0 {
```

Comment:
This comment doesn't match the new if statement. I think the if statement is backwards but one of them should be flipped.

Reply:
flipped the comment.
I'm pruning odds instead of evens so the first machine will not be pruned. E.g if narrowing down on a a range of 10K instructions starting from step 1M - you never want to prune the machine @1M because you might need it later and don't want to recalculate it.